graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/DgsRolloutFlags.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs
+
+data class DgsRolloutFlags(
```

Comment:
I'm not particularly happy with having to introduce this, but I can't think of a way to inject this config into `DefaultDgsFederationResolver` (and subclasses) without it.

Reply:
Instead of creating a bean for DgsRolloutFlags, you can create a bean for the DgsFederationResolver. This bean will be injected into the DgsSchemaProvider, so you can pass in the config to the DgsFederationResolver directly when creating the bean there in the DgsAutoConfiguration.

Reply:
I think you could just pass the config setting as a boolean flag into the DgsSchemaProvider and then pass that through to the federation resolver.

Reply:
Unfortunately I don't think so, since DGSes can (and do) subclass `DefaultDgsFederationResolver` and then rely on the `@Autowired` annotations instead of the constructor. However, I think Paul's suggestion is a nice sidestep to this problem.

Reply:
Once concern with introducing a flag in the annotation might make it harder to deprecate, since it is now kind of a feature rather than a bug fix. But we could also flip the default for the flag I guess in the future in a breaking release.

Reply:
Yeah my suggestion is only good if *is* like a feature. If it's really considered a bug fix I agree it shouldn't be part of the API. If it's in the annotation, it will need to stay.

Reply:
Yeah, I've swapped back to the property as we're really viewing this as a bugfix, i.e. bringing it in line with input parsing.